main.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    arr = to_a if self.class == Hash || self.class == Range
+    while i < size
+      if self.class == Hash || self.class == Range
+        yield(arr.at(i))
+      else
+        yield(at(i))
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index(*arg)
+    return to_enum(:my_each_with_index) unless block_given? && arg.empty?
+
+    index = 0
+    arr = self
+    tarr = arr.to_a
+    i = 0
+    while i < arr.size
+      yield(tarr[i], index)
+      index += 1
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    item = []
+    arr = self
+    tarr = []
+    if arr.class == Hash
+      arr.my_each { |key, _value| tarr << key }
+    else
+      tarr = arr.to_a
+    end
+    i = 0
+    while i < tarr.size
+      yield(tarr[i]) ? item << tarr[i] : nil
+      i += 1
+    end
+    item
+  end
+
+  def my_all?(*args)
+    flag = 0
+    i = 0
+    arr = self
+    is_a?(Range) || is_a?(Hash) ? arr = arr.to_a : nil
+
+    case args.empty?
+    when true
+      if block_given?
+        blk = yield(arr.at(i))
+        arr.my_each { flag += 1 if blk == true }
+      else
+        arr.my_each { flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      arr.my_each { flag += 1 if i.is_a?(args[0]) }
+    end
+    return true if flag == size
+    return false if flag != size
+  end
+
+  def my_any?(*args)
+    flag = 0
+    case args.empty?
+    when true
+      if block_given?
+        my_each { |i| flag += 1 if yield(i) }
+      else
+        my_each { |i| flag += 1 unless i.nil? || i == false }
+      end
+    when false
+      my_each { |i| flag += 1 if i.is_a?(args[0]) }
+    end
+    return true if flag.positive?
+    return false if flag <= 0
+  end
```

Comment:
# my_any? 
1. [ ] When a Regex is passed as an argument returns true if any of the collection matches the Regex

     Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)

     TypeError:
       class or module required
##
# my_any? 
2. [ ] When a Regex is passed as an argument returns false if none of the collection matches the Regex

     Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)

     TypeError:
       class or module required
##
# my_any? 
3. [ ] When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

     Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

     TypeError:
       class or module required
##   
# my_any? 
4. [ ] When a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

      TypeError:
        class or module required